controlplane/kubeadm/internal/filters.go

```diff
@@ -215,15 +215,28 @@ func matchInitOrJoinConfiguration(machineConfig *bootstrapv1.KubeadmConfig, kcp
 	// *Note* This assumes that newly added default values never
 	// introduce a semantic difference to the unset value.
 	// But that is something that is ensured by our API guarantees.
-	bootstrapv1.DefaultKubeadmConfigSpec(kcpConfig)
-	bootstrapv1.DefaultKubeadmConfigSpec(&machineConfig.Spec)
+	defaultKubeadmConfigSpec(kcpConfig)
+	defaultKubeadmConfigSpec(&machineConfig.Spec)
 
 	// cleanups all the fields that are not relevant for the comparison.
 	cleanupConfigFields(kcpConfig, machineConfig)
 
 	return reflect.DeepEqual(&machineConfig.Spec, kcpConfig)
 }
 
+// defaultKubeadmConfigSpec defaults a KubeadmConfigSpec.
+func defaultKubeadmConfigSpec(r *bootstrapv1.KubeadmConfigSpec) {
```

Comment:
Same here, we have to use  the same defaulting as in the KubeadmConfig webhook

Reply:
It is absolutely crucial that this code does not diverge 

If this runs out of sync KCP could trigger rollouts when none should be happening because it anticipates KubeadmConfigSpec defaulting incorrectly